.github/workflows/mypy.yml

```diff
@@ -72,3 +72,21 @@ jobs:
     - name: Run Mypy
       run: |
         spin mypy
+  # This job checks for suspicious C API usage that may break nogil compatibility.
+  # It scans PR diffs for dangerous borrowed-referenced calls and fails if any are found.
+  lint-capi:
```

Comment:
This doesn't need a new job, we already have two lint jobs. And I think the checks are already run there by default when `tools/linter.py` is invoked, here (please verify):

https://github.com/numpy/numpy/blob/d32370539ea5aa93100842a98cf21d55667b6041/.github/workflows/linux.yml#L47

So this job can simply be removed?